mllib/src/main/scala/org/apache/spark/ml/feature/RFormulaParser.scala

```diff
@@ -298,7 +299,10 @@ private[ml] object RFormulaParser extends RegexParsers {
   private val expr = (sum | term)
 
   private val formula: Parser[ParsedRFormula] =
-    (label ~ "~" ~ expr) ^^ { case r ~ "~" ~ t => ParsedRFormula(r, t.asTerms.terms) }
+    (label ~ "~" ~ expr) ^^ {
+      case r ~ "~" ~ t => ParsedRFormula(r, t.asTerms.terms)
```

Comment:
For future reference, a slightly cleaner way to express this in a warning free way is:

```
((label <~ "~") ~ expr ) ^^ {
  case r ~ t => ParsedRFormula(r, t.asTerms.terms)
}
```

It also avoids duplicating the `"~"` token in the parser and the action.

Reply:
Feel free to make a PR, @retronym . :)